uxarray/grid/grid.py

```diff
@@ -1221,3 +1240,151 @@ def isel(self, **dim_kwargs):
+
+
+            else:
+                # Normal Face
+                for j in range(face_edges_connectivity_cartesian.shape[0]):
+                    edge = face_edges_connectivity_cartesian[j]
+                    # Skip the dummy edge
+                    if np.any(edge == INT_FILL_VALUE):
+                        continue
+
+                    # Get the cartesian coordinates of the edge
+                    n1_cart = [self.node_x[edge[0]], self.node_y[edge[0]], self.node_z[edge[0]]]
+                    n2_cart = [self.node_x[edge[1]], self.node_y[edge[1]], self.node_z[edge[1]]]
+
+                    # Insert the endpoints of the edge into the latlonbox, we only need to insert the first point
+                    # as the second point will be the first point of the next edge
+                    pt_lat_rad = np.deg2rad(self.node_lat.values[edge[0]])
+                    pt_lon_rad = np.deg2rad(self.node_lon.values[edge[0]])
+
+                    # Now let's see if the extreme latitudes of the edge is not the same as the latitudes of the nodes
+                    # If it is not, we need to insert the extreme latitudes into the latlonbox
+                    lat_max = extreme_gca_latitude(n1_cart, n2_cart, 'max')
+                    lat_min = extreme_gca_latitude(n1_cart, n2_cart, 'min')
+
+                    if np.isclose(self.np.deg2rad(self.node_lat.values[edge[0]]), lat_max, atol=ERROR_TOLERANCE)\
+                            or np.isclose(self.np.deg2rad(self.node_lat.values[edge[1]]), lat_max, atol=ERROR_TOLERANCE):
+
+                        temp_latlon_array[face_idx] = _insert_pt_in_latlonbox(
+                            copy.copy(temp_latlon_array[face_idx]),
+                            np.array([lat_max, pt_lon_rad]))
+
+                    elif np.isclose(self.np.deg2rad(self.node_lat.values[edge[0]]), lat_min, atol=ERROR_TOLERANCE)\
+                            or np.isclose(self.np.deg2rad(self.node_lat.values[edge[1]]), lat_min, atol=ERROR_TOLERANCE):
+
+                        temp_latlon_array[face_idx] = _insert_pt_in_latlonbox(
+                            copy.copy(temp_latlon_array[face_idx]),
+                            np.array([lat_min, pt_lon_rad]))
+
+                    else:
+                        temp_latlon_array[face_idx] = _insert_pt_in_latlonbox(
+                            copy.copy(temp_latlon_array[face_idx]), np.array([pt_lat_rad, pt_lon_rad]))
+
+            assert temp_latlon_array[face_idx][0][0] != temp_latlon_array[face_idx][0][1]
+            assert temp_latlon_array[face_idx][1][0] != temp_latlon_array[face_idx][1][1]
+            lat_array = temp_latlon_array[face_idx][0]
+
+            # Now store the latitude intervals in the DataFrame
+            intervals_list = intervals_list.append({'lower_lat': lat_array[0], 'upper_lat': lat_array[1], "face_idx": face_idx})
+
+        lattitude_intervals_df = pd.DataFrame(intervals_list)
+        self._bounds = temp_latlon_array
```

Comment:
Hi @philipc2 ,
This `Self._bounds` should be a Xarray storing the latitude longitude bound for each face.

Reply:
Yes, should be quite similar in form to how we store constructure connectivity or coordinate variables.

A reference below 

https://github.com/UXARRAY/uxarray/blob/871e270b3ef3fe56c80d7d7e2d51910ccd79280f/uxarray/grid/connectivity.py#L169-L180